src/Lykke.Sdk/LykkeApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,119 @@
+        {
+            var log = (ILog)app.ApplicationServices.GetService(typeof(ILog));
+
+            try
+            {
+                var options = new LykkeAppOptions();
+                optionBuilder(options);
+
+                if (string.IsNullOrWhiteSpace(options.AppName))
+                    throw new ApplicationException("Application name is required.");
+
+                if (string.IsNullOrWhiteSpace(options.Version))
+                    throw new ApplicationException("Version is required.");
+
+                var appLifetime = (IApplicationLifetime) app.ApplicationServices.GetService(typeof(IApplicationLifetime));
+                var configurationRoot = (IConfigurationRoot) app.ApplicationServices.GetService(typeof(IConfigurationRoot));
+                var monitoringSettings = (IReloadingManager<MonitoringServiceClientSettings>) app.ApplicationServices.GetService(typeof(IReloadingManager<MonitoringServiceClientSettings>));
+                
+                var startupManager = (IStartupManager) app.ApplicationServices.GetService(typeof(IStartupManager));
+                var shutdownManager = (IShutdownManager) app.ApplicationServices.GetService(typeof(IShutdownManager));
+
+                appLifetime.ApplicationStarted.Register(() =>
+                {
+                    try
+                    {
+                        startupManager?.StartAsync().GetAwaiter().GetResult();
+
+                        log.WriteMonitor("StartApplication", null, "Application started");
+
+                        if (!options.IsDebug)
+                        {
+                            if (monitoringSettings.CurrentValue == null)
+                                throw new ApplicationException("Monitoring settings is not provided.");
+
+                            AutoRegistrationInMonitoring.RegisterAsync(configurationRoot, monitoringSettings.CurrentValue.MonitoringServiceUrl, log).GetAwaiter().GetResult();
+                        }
+
+                    }
+                    catch (Exception ex)
+                    {
+                        log.WriteFatalError("StartApplication", "", ex);
+                        throw;
+                    }
+                });
+
+                appLifetime.ApplicationStopping.Register(() =>
+                {
+                    try
+                    {
+                        shutdownManager?.StopAsync().GetAwaiter().GetResult();
+                    }
+                    catch (Exception ex)
+                    {
+                        log?.WriteFatalError("StopApplication", "", ex);
+
+                        throw;
+                    }
+                });
+
+                appLifetime.ApplicationStopped.Register(() =>
+                {
+                    try
+                    {
+                        log?.WriteMonitor("StopApplication", null, "Terminating");
```

Comment:
Is https://github.com/LykkeCity/Lykke.Sdk/pull/1/files#diff-38921aaf7384c53525df11deeeb9f09cR32 will be called before this handler or after? Looks like before, then this is incorrect. May be will be better to keep only one handler for ```ApplicationStopped``` event in ```BuildServiceProvider``` and place all necessary code there

Reply:
There was one handler, but i decided to split, and yes, it's not better solution.